pkg/rslog/debug/debug.go

```diff
@@ -104,32 +104,30 @@ func Enabled(region ProductRegion) bool {
 type DebugLogger interface {
 	Enabled() bool
 	Debugf(msg string, args ...interface{})
+	Tracef(msg string, args ...interface{})
 	WithFields(fields rslog.Fields) DebugLogger
 	WithSubRegion(subregion string) DebugLogger
 }
 
 type debugLogger struct {
 	rslog.Logger
-	region ProductRegion
+	region  ProductRegion
+	enabled bool
 }
 
 // NewDebugLogger returns a new logger which includes
 // the name of the debug region at every message.
-func NewDebugLogger(region ProductRegion, lgrIn rslog.Logger) *debugLogger {
-	lgr := lgrIn.Copy()
-
-	if Enabled(region) {
-		lgr.SetLevel(rslog.DebugLevel)
-		lgr.SetReportCaller(true)
-	}
+func NewDebugLogger(region ProductRegion) *debugLogger {
+	lgr := rslog.DefaultLogger()
```

Comment:
It is looking good to me. One issue we currently have, noticed by @eduardocfalcao , is that sometimes `NewDebugLogger()` is called before the default logger is properly initialized(due to order of package imports in Connect), leaving a logger instance that is not in sync with the configuration set.

We are thinking this could be solved if `rslog.DefaultLogger()` returns a pointer to the default logger, instead of the value. WDYT?

Reply:
Yes, that's one of the primary purposes of this PR. 😁 I still need to do some further testing, but I think the singleton default logger (that cannot be replaced) will solve that problem.

Reply:
Oh, yes, returning the default logger instead of a copy should solve that problem, I got confused 😅